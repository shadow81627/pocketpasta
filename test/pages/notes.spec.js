import Component from '@/pages/notes/index.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';

const localVue = createLocalVue();

localVue.use(Vuetify);
localVue.use(VueRouter);

const router = new VueRouter();

const factory = () =>
  shallowMount(Component, {
    localVue,
    router,
    $pouch: {
      changes: () => ({ on: () => ({}) }),
      post: () => ({}),
      put: () => ({}),
      upsert: () => ({}),
      find: () => [],
    },
    mocks: {
      $fetchState: {},
      $pouch: {
        changes: () => ({ on: () => ({}) }),
        post: () => ({}),
        put: () => ({}),
        upsert: () => ({}),
        find: () => [],
      },
    },
  });

describe('Notes page', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('deleteItem', async () => {
    const wrapper = factory();
    expect(await wrapper.vm.deleteItem({ _id: 'test' })).toBe(undefined);
  });
  test('save', async () => {
    const wrapper = factory();
    expect(await wrapper.vm.save({})).toBe(undefined);
  });

  test('fetch', async () => {
    const wrapper = factory();
    expect(await wrapper.vm.$options.fetch()).toEqual(undefined);
    expect(wrapper.vm.items).toEqual([]);
  });

  test('content', () => {
    const wrapper = factory();
    wrapper.vm.content = 1;
    expect(wrapper.vm.content).toEqual(undefined);
  });

  test('head', () => {
    expect(Component.head()).toBeTruthy();
  });
});
