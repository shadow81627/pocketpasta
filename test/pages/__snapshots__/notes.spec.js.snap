// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes page renders properly 1`] = `
"<v-container-stub tag=\\"div\\">
  <v-row-stub tag=\\"div\\">
    <v-col-stub tag=\\"div\\">
      <h1>Notes</h1>
    </v-col-stub>
  </v-row-stub>
  <v-row-stub tag=\\"div\\">
    <v-col-stub tag=\\"div\\"><span><tiptap-vuetify-stub value=\\"\\" extensions=\\"function e(e){return t.call(this,e,r.History)||this},function e(e){return t.call(this,e,r.Blockquote)||this},function n(e){return t.call(this,e,r.Link)||this},function e(e){return t.call(this,e,r.Underline)||this},function e(e){return t.call(this,e,r.Strike)||this},function e(e){return t.call(this,e,r.Italic)||this},function e(e){return t.call(this,e,r.ListItem)||this},function e(e){return t.call(this,e,r.BulletList)||this},function e(e){return t.call(this,e,r.OrderedList)||this},function e(e){return t.call(this,e,r.TodoList)||this},function e(e){return t.call(this,e,td)||this},function e(e){return t.call(this,e,r.Heading)||this},[object Object],function e(e){return t.call(this,e,r.Bold)||this},function n(e){return t.call(this,e,r.Link)||this},function e(e){return t.call(this,e,r.Code)||this},function e(e){return t.call(this,e,r.CodeBlock)||this},function n(e){return t.call(this,e,r.Image)||this},function e(e){return t.call(this,e,r.HorizontalRule)||this},function e(e){return t.call(this,e,null)||this},function e(e){return t.call(this,e,r.HardBreak)||this}\\" placeholder=\\"Write something â€¦\\" cardprops=\\"[object Object]\\" outputformat=\\"html\\" toolbarattributes=\\"[object Object]\\" editorproperties=\\"[object Object]\\" nativeextensions=\\"\\" type=\\"card\\"></tiptap-vuetify-stub></span></v-col-stub>
  </v-row-stub>
</v-container-stub>"
`;
